endpoints.go

```diff
@@ -33,3 +43,389 @@ func showHealth(w http.ResponseWriter, r *http.Request) {
 	healthInfo := models.Health{Database: dbStatus, RabbitMQ: viper.GetString("dummy_health_rabbitmq")}
 	json.NewEncoder(w).Encode(healthInfo)
 }
+
+//Find survey by reference, short name, long name, or any combination of the three
+func getSurvey(w http.ResponseWriter, r *http.Request) {
+
+    if db == nil {
+        w.WriteHeader(http.StatusInternalServerError)
+        errorString := models.Error{
+            Error: "Database connection could not be found",
+        }
+        json.NewEncoder(w).Encode(errorString)
+        return
+    }
+
+    queryParams := r.URL.Query()
+    if len(queryParams) == 0 {
+        w.WriteHeader(http.StatusBadRequest)
+        errorString := models.Error{
+            Error: "No query parameters provided for search",
+        }
+        json.NewEncoder(w).Encode(errorString)
+        return
+    }
+
+    var sb strings.Builder
+    sb.WriteString(" WHERE 1=1")
+    for params := range queryParams {
+        switch params {
+        case "surveyRef":
+            sb.WriteString(" AND survey_ref='")
+            sb.WriteString(queryParams.Get("surveyRef"))
+            sb.WriteString("'")
+        case "shortName":
+            sb.WriteString(" AND short_name='")
+            sb.WriteString(queryParams.Get("shortName"))
+            sb.WriteString("'")
+        case "longName":
+            sb.WriteString(" AND long_name='")
+            sb.WriteString(queryParams.Get("longName"))
+            sb.WriteString("'")
+        default:
+            w.WriteHeader(http.StatusBadRequest)
+            errorString := models.Error{
+                Error: "Invalid query parameter " + params,
+            }
+            json.NewEncoder(w).Encode(errorString)
+            return
+        }
+    }
+
```

Comment:
I believe this still is open to SQL injection attacks, rather than building a string and executing that you either need to pass the values as multiple parameters the db.Query or preferably build up a paramaterized string and then prepare it using the parameteres you've got the url.